gen/bbmri_negotiator

```diff
@@ -0,0 +1,169 @@
+our $A_G_ADM_COLLECTION_ID;       *A_G_ADM_COLLECTION_ID =     \'urn:perun:group:attribute-def:def:admCollectionID';
+our $A_G_ADM_BIOBANK_ID;          *A_G_ADM_BIOBANK_ID =        \'urn:perun:group:attribute-def:def:admBiobankID';
+our $A_G_ADM_NATIONAL_NODE_ID;    *A_G_ADM_NATIONAL_NODE_ID =  \'urn:perun:group:attribute-def:def:admNationalNodeID';
+our $A_G_ADM_NETWORK_ID;          *A_G_ADM_NETWORK_ID =        \'urn:perun:group:attribute-def:def:admNetworkID';
+
+our $STATUS_VALID;                *STATUS_VALID =            \'VALID';
+
+our $ROLE_COLLECTION;       *ROLE_COLLECTION =   \'COLLECTION';
+our $ROLE_BIOBANK;          *ROLE_BIOBANK =      \'BIOBANK';
+our $ROLE_NN;               *ROLE_NN =           \'NATIONAL_NODE';
+our $ROLE_NETWORK;          *ROLE_NETWORK =      \'NETWORK';
+
+our $ROLE_ADM_COLLECTION;       *ROLE_ADM_COLLECTION =   \'ADM_COLLECTION';
+our $ROLE_ADM_BIOBANK;          *ROLE_ADM_BIOBANK =      \'ADM_BIOBANK';
+our $ROLE_ADM_NN;               *ROLE_ADM_NN =           \'ADM_NATIONAL_NODE';
+our $ROLE_ADM_NETWORK;          *ROLE_ADM_NETWORK =      \'ADM_NETWORK';
+
+our $userStruc = {};
+our $USER_EMAIL = "EMAIL";
+our $USER_NAME = "NAME";
+
+my $fileUsers = $DIRECTORY . "negotiator.json";
+
+foreach my $resourceId ( $data->getResourceIds() ) {
+	foreach my $groupId ( $data->getGroupIdsForResource( resource => $resourceId ) ) {
+		processGroups $groupId, $resourceId;
+	}
+}
+
+# PREPARE USERSDATA TO JSON
+my @users;
+foreach my $uid (sort keys %$userStruc) {
+	my $user = {};
+	$user->{"id"} = $uid;
+	$user->{"displayName"} = $userStruc->{$uid}->{$USER_NAME};
+	$user->{"mail"} = $userStruc->{$uid}->{$USER_EMAIL};
+	$user->{"membership"} = {};
+	$user->{"membership"}->{"collections"} = $userStruc->{$uid}->{$ROLE_COLLECTION};
+	$user->{"membership"}->{"biobanks"} = $userStruc->{$uid}->{$ROLE_BIOBANK};
+	$user->{"membership"}->{"national_nodes"} = $userStruc->{$uid}->{$ROLE_NN};
+	$user->{"membership"}->{"networks"} = $userStruc->{$uid}->{$ROLE_NETWORK};
+	$user->{"administrator"} = {};
+	$user->{"administrator"}->{"collections"} = $userStruc->{$uid}->{$ROLE_ADM_COLLECTION};
+	$user->{"administrator"}->{"biobanks"} = $userStruc->{$uid}->{$ROLE_ADM_BIOBANK};
+	$user->{"administrator"}->{"national_nodes"} = $userStruc->{$uid}->{$ROLE_ADM_NN};
+	$user->{"administrator"}->{"networks"} = $userStruc->{$uid}->{$ROLE_ADM_NETWORK};
+
+	push @users, $user;
+}
+
+# PRINT USERS TO JSON
+open FILE_USERS,">$fileUsers" or die "Cannot open $fileUsers: $! \n";
+binmode(FILE_USERS);
+print FILE_USERS JSON::XS->new->utf8->pretty->canonical->encode(\@users);
+close (FILE_USERS) or die "Cannot close $fileUsers: $! \n";
+
+perunServicesInit::finalize;
+
+## creates users structure
+sub processGroups {
+	my $groupId = shift;
+	my $resourceId = shift;
+
+	my $roleIdValue;
+	my $groupRole;
+
+	if ($data->getGroupAttributeValue( group => $groupId, attrName => $A_G_COLLECTION_ID )) {
```

Comment:
I'm assuming a group can only be one of these (biobank, network, etc.). In which case, wouldn't a check that only one of the attributes is set be nice? Or are the branches in this order because of "priority" of the attributes?